bin/Macronify/Macronify.cpp

```diff
@@ -5,43 +5,33 @@
 // LICENSE file found in the root directory of this source tree.
 
 #include "ParseAST.hpp"
-#include "macroni/Common/CodeGenDriverSetup.hpp"
+#include "macroni/Common/GenerateModule.hpp"
 #include "macroni/Dialect/Macroni/MacroniDialect.hpp"
 #include "macroni/Translation/Macroni/MacroniVisitor.hpp"
 #include "macroni/Translation/Macroni/PastaMetaGenerator.hpp"
-#include "pasta/AST/AST.h"
-#include "vast/CodeGen/CodeGenDriver.hpp"
-#include <clang/AST/ASTContext.h>
-#include <iostream>
-#include <memory>
-#include <mlir/IR/DialectRegistry.h>
-#include <mlir/InitAllDialects.h>
+#include "vast/Util/Common.hpp"
+#include <cstdlib>
+#include <llvm/Support/raw_ostream.h>
+#include <mlir/IR/Verifier.h>
 
 int main(int argc, char **argv) {
   auto maybe_ast = pasta::parse_ast(argc, argv);
   if (!maybe_ast.Succeeded()) {
-    std::cerr << maybe_ast.TakeError() << '\n';
+    llvm::errs() << maybe_ast.TakeError() << '\n';
     return EXIT_FAILURE;
   }
-
-  auto pctx = maybe_ast.TakeValue();
+  auto &pctx = maybe_ast.Value();
   auto &actx = pctx.UnderlyingAST();
-  auto driver =
-      macroni::generate_codegen_driver<macroni::macroni::MacroniDialect,
-                                       macroni::pasta_meta_generator>(actx);
-
-  // Add the macroni visitor first. This ensures that our visitor will try to
-  // lower stmts down to the Macroni dialect first before VAST's dialect.
-  driver->push_visitor(std::make_unique<macroni::macroni_visitor>(
-      pctx, driver->mcontext(), driver->get_codegen_builder(),
-      driver->get_meta_generator(), driver->get_symbol_generator(),
-      vast::cg::visitor_view(driver->get_visitor_stack())));
-
-  vast::cg::setup_default_visitor_stack(*driver);
+  auto mctx = vast::mcontext_t();
```

Comment:
Is using `vast::cg::mk_mcontext()` not possible here? I think we planned to adding more logic to it, so may be handy to use on your side (you will get those changes for free of work then).

Reply:
Good point, I'll just call `vast::cg::mk_mcontext()` then